draft-irtf-cfrg-frost.md

```diff
@@ -1187,16 +1187,28 @@ an invalid signature. To avoid this denial of service, implementations may wish
 to define a mechanism where messages are authenticated, so that cheating players
 can be identified and excluded.
 
+## Input Message Hashing {#pre-hashing}
+
+FROST signatures do not pre-hash message inputs. This means that the entire message
+must be known in advance of invoking the signing protocol. Applications can apply
+pre-hashing in settings where storing the full message is prohibitively expensive.
+In such cases, pre-hashing MUST use a collision-resistant hash function with a security
+level commensurate with the security in inherent to the ciphersuite chosen. It is
+RECOMMENDED that applications which choose to apply pre-hashing use the hash function
+(`H`) associated with the chosen ciphersuite in a manner similar to how `H4` is defined.
```

Comment:
Any reason to not recommend using H4 explicitly to pre-hash?

Reply:
Yeah, it doesn't seem safe to repurpose the internal random oracles for things outside of the protocol.

Reply:
If we can't/shouldn't 'just' use H4 for prehashing as well we should explain how to use H in a 'manner similar to H4' that is not just 'the same' as H4. Does this mean a different domain separator? If that's it we can say that but I am uncertain. @chelseakomlo does prehashing need to be domain separated from regular message hashing in the security model? I have not dug into this much

Reply:
Yeah, I think it would need a different domain separator to be different from H4.

Reply:
RFC 8032 does that: for regular Ed25519 it's H(msg), for prehashed is H(prefix + hash). For Ed448 it's H(prefix + hash) with different prefixes for prehashed and non-prehashed.

Reply:
I added an example that describes how this pre-hash might be constructed. Please take a look, @dconnolly.